src/hooks/useSearchHighlightAndScroll.ts

```diff
@@ -3,43 +3,51 @@ import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+ * Hook used to trigger a search when a new transaction or report action is added and handle highlighting and scrolling.
  */
-function useSearchHighlightAndScroll({searchResults, transactions, previousTransactions, queryJSON, offset}: UseSearchHighlightAndScroll) {
+function useSearchHighlightAndScroll({searchResults, transactions, previousTransactions, reportActions, previousReportActions, queryJSON, offset}: UseSearchHighlightAndScroll) {
     // Ref to track if the search was triggered by this hook
     const triggeredByHookRef = useRef(false);
     const searchTriggeredRef = useRef(false);
     const previousSearchResults = usePrevious(searchResults?.data);
     const [newSearchResultKey, setNewSearchResultKey] = useState<string | null>(null);
-    const highlightedTransactionIDs = useRef<Set<string>>(new Set());
+    const highlightedIDs = useRef<Set<string>>(new Set());
     const initializedRef = useRef(false);
+    const isChat = queryJSON.type === CONST.SEARCH.DATA_TYPES.CHAT;
 
-    // Trigger search when a new transaction is added
+    // Trigger search when a new report action is added while on chat or when a new transaction is added for the other search types.
     useEffect(() => {
         const previousTransactionsLength = previousTransactions && Object.keys(previousTransactions).length;
         const transactionsLength = transactions && Object.keys(transactions).length;
 
-        // Return early if search was already triggered or there's no change in transactions length
-        if (searchTriggeredRef.current || previousTransactionsLength === transactionsLength) {
+        const reportActionsLength = reportActions && Object.values(reportActions).reduce((sum, curr) => sum + Object.keys(curr ?? {}).length, 0);
+        const prevReportActionsLength = previousReportActions && Object.values(previousReportActions).reduce((sum, curr) => sum + Object.keys(curr ?? {}).length, 0);
+        // Return early if search was already triggered or there's no change in current and previous data length
+        if (searchTriggeredRef.current || (!isChat && previousTransactionsLength === transactionsLength) || (isChat && reportActionsLength === prevReportActionsLength)) {
             return;
         }
+        const newTransactionAdded = transactionsLength && typeof previousTransactionsLength === 'number' && transactionsLength > previousTransactionsLength;
+        const newReportActionAdded = reportActionsLength && typeof prevReportActionsLength === 'number' && reportActionsLength > prevReportActionsLength;
 
-        // Check if a new transaction was added
-        if (transactionsLength && typeof previousTransactionsLength === 'number' && transactionsLength > previousTransactionsLength) {
+        // Check if a new transaction or report action was added
+        if ((!isChat && !!newTransactionAdded) || (isChat && !!newReportActionAdded)) {
```

Comment:
Coming from https://github.com/Expensify/App/issues/57733 , we’ve added the `isChat` condition to `newReportActionAdded`, which prevents the Search API from being triggered when a self-DMed expense is changed to a workspace expense. More context here: https://github.com/Expensify/App/issues/57733#issuecomment-2696977077